src/map/script.cpp

```diff
@@ -24235,6 +24236,37 @@ BUILDIN_FUNC(unloadnpc) {
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/**
+ * Duplicate a NPC.
+ * Return the duplicate name on success.
+ * duplicate(<"NPC name">{,<"Duplicate NPC name">});
+ */
+BUILDIN_FUNC(duplicate)
+{
+	npc_data* dnd = npc_name2id(script_getstr(st, 2));
+
+	if (dnd == nullptr) {
+		ShowError("buildin_duplicate: No such NPC '%s'.\n", script_getstr(st, 2));
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	npc_data* nd = nullptr;
```

Comment:
```suggestion
	npc_data* nd;
```
Doesn't really need to be assigned anything here since it will be done right below.